searx/answerers/__init__.py

```diff
@@ -23,28 +21,32 @@ def load_answerers():
     return answerers
 
 
-def get_answerers_by_keywords(answerers):
-    by_keyword = defaultdict(list)
-    for answerer in answerers:
-        for keyword in answerer.keywords:
-            for keyword in answerer.keywords:
-                by_keyword[keyword].append(answerer.answer)
-    return by_keyword
-
+class Answerers():
```

Comment:
Do we need to create a class for this? Shawn’s mocks assuming it’s just a function.

Reply:
Shawn’s mocks assume it’s just a function.*

Reply:
well I'm making the loader pluggable and it is used by all three functions. I didn't want to keep state in the file/module as a variable

Reply:
Looking at his commit, it seems he will just need to change the import and the line it's used at.

```python
from searx.answerers import Answerers, file_loader
# ...
Answerers(file_loader).ask(...)
```

vs.

```python
from searx.answerers import ask
# ...
ask(...)
```